lib/shadow.ml

```diff
@@ -31,9 +31,17 @@ let sp_flag     = field shadow_t "sp_flag" ulong
 
 let () = seal shadow_t
 
+let ptr_char_to_string = coerce (ptr char) string
+
+let string_to_char_array s =
+  let len = String.length s in
+  let buf = CArray.make char ~initial:'\x00' len in
+  String.iteri (fun idx c -> CArray.set buf idx c) s;
+  buf
+
```

Comment:
Can we avoid the duplication of these functions?

Reply:
@lindig was asking the same. We can but at the price of exposing them to the outer world. They are so small that I prefer to keep them hidden in the modules instead